influxql/iterator.gen.go

```diff
@@ -403,6 +403,10 @@ func (itr *floatLimitIterator) Next() *FloatPoint {
 
 		// Read next point if we're beyond the limit.
 		if itr.opt.Limit > 0 && (itr.n-itr.opt.Offset) > itr.opt.Limit {
+			// If there's no interval and no groups then simply exit.
+			if itr.opt.Interval.Duration == 0 && len(itr.opt.Dimensions) == 0 {
```

Comment:
Fixed. Also, I made the change to `iterator.gen.go` instead of `iterator.gen.go.tmpl`. :facepalm: Fixed that too.